ExampleMod/Common/GlobalPylons/ExampleGlobalPylon.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.GameContent;
+using Terraria.Map;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Common.GlobalPylons
+{
+	/// <summary>
+	/// An example and showcase of some of the hooks within the GlobalPylon class, which allow us to change functionality
+	/// for any type of pylon that we want. The changes we make in this class may not be the most practical, as they are more-so
+	/// showing what you can do with the hooks.
+	/// </summary>.
+	public class ExampleGlobalPylon : GlobalPylon
+	{
+		public override bool? ValidTeleportCheck_PreNPCCount(TeleportPylonInfo pylonInfo, ref int defaultNecessaryNPCCount) {
+			//Since we have the capabilities, we can allow players to teleport to any pylon even if there are no NPCs there during the night time.
+			if (!Main.dayTime) {
+				defaultNecessaryNPCCount = 0;
+			}
+
+			//Since we aren't preventing anything and just changing the NPC count, we can just return what the default method returns, which is null (AKA vanilla behavior)
+			return base.ValidTeleportCheck_PreNPCCount(pylonInfo, ref defaultNecessaryNPCCount);
+		}
+
+		public override bool PreDrawMapIcon(ref MapOverlayDrawContext context, ref string mouseOverText, ref TeleportPylonInfo pylonInfo, ref bool isNearPylon, ref Color drawColor, ref float deselectedScale, ref float selectedScale) {
+			//What if we want to change the color of all of the map icons?
+			//If we aren't near a pylon, we're going to shift the color of all pylon icons to being more red
+			if (!isNearPylon) {
+				drawColor = Color.Lerp(drawColor, Color.Red, 0.75f);
+			}
+
+			//Since we aren't actually preventing the drawing of any map icons, we can just return the default value, which in this case is null (AKA vanilla behavior)
+			return base.PreDrawMapIcon(ref context, ref mouseOverText, ref pylonInfo, ref isNearPylon, ref drawColor, ref deselectedScale, ref selectedScale);
+		}
+
+		public override bool? PreCanPlacePylon(int x, int y, int tileType, byte pylonType) {
+			//What if we want to override the functionality for pylon placement?
+			//For example, let's always allow the players to place universal pylons, even if they already exist in the world:
+			if (pylonType == (byte)TeleportPylonType.Victory) {
+				return true;
+			}
+
+			return base.PreCanPlacePylon(x, y, tileType, pylonType);
+		}
+
+		public override bool? ValidTeleportCheck_PreBiomeRequirements(TeleportPylonInfo pylonInfo, SceneMetrics sceneData) {
+			//What if we want to do something based on the type of pylon in particular? Well all we have to do is check the pylon's type!
+			//Let's swap around the biome requirements for the Jungle Pylon and Snow Pylon, for example:
+			TeleportPylonType pylonType = pylonInfo.TypeOfPylon;
+
+			if (pylonType == TeleportPylonType.Jungle) {
```

Comment:
A better example would probably be to allow jungle pylon in the snow (as well as jungle). With this example, the false return will cancel out any other mod's attempt to make the jungle pylon more permissible